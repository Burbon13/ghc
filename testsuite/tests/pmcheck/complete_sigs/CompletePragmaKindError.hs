{-# LANGUAGE PatternSynonyms #-}
{-# LANGUAGE ViewPatterns #-}
{-# LANGUAGE TypeFamilies #-}
{-# LANGUAGE MultiParamTypeClasses #-}
{-# LANGUAGE FunctionalDependencies #-}
{-# LANGUAGE FlexibleInstances #-}

import Data.Functor.Identity

newtype App f x = App (f x)

class WrappedIn s a | s -> a where
  unwrap :: s -> a

instance WrappedIn (App f a) (f a) where
  unwrap (App fa) = fa

pattern Unwrapped :: WrappedIn s a => a -> s
pattern Unwrapped x <- (unwrap -> x)

{-# COMPLETE Unwrapped :: App t #-}

boom :: App Identity t -> Bool
boom (Unwrapped (Identity _)) = True

main :: IO ()
main = print ":)"
