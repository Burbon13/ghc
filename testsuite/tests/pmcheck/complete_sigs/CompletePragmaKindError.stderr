
CompletePragmaKindError.hs:21:27: error:
    • Expecting one more argument to ‘App t’
      Expected a type, but ‘App t’ has kind ‘* -> *’
    • In the type ‘App t’
      In the scrutinee type of a COMPLETE pragma: App t
